### DISCUSSION TITLE
Added support for custom board (STM32H750B-DK) - not working - how to troubleshoot?

### DISCUSSION THREAD

**Discussion by LAPSyLAB** [micropython/micropython#10290] [STM32 / Pyboard]
Hello,

we have a lot of boards of this type used education process and would like to add MicroPyhon support for this board.

I've done to the best of my knowledge from examples and docs found so far. I get two bin files, program each in internal and QSPI flash, but board doesn't respond in any way. 

Now I'm trying to figure out what is wrong.

- is there any way I can debug or set somekind of debug prints on VCOM port (on this board USART3 is connected as VCOM to usb stlink connection). HW works in other projects with other code ?

- this is what I've done so far (probably have some obvious mistakes in configrations... So I'm kindly asking for a quick review and advice how to configure correctly. 

I'm new to Micropython, so this is layman's work. :-)

Thanks in advance,
regards,
Rob.

[STM32H750B_DK.zip](https://github.com/micropython/micropython/files/10277038/STM32H750B_DK.zip)

**Comment by dpgeorge** [comment]
The H750 is tricky because it only has 128k of internal flash, so firmware larger than this must go in external QSPI (XIP) flash.  Then the part of the firmware in the internal 128k would need to enable the XIP flash before any code executes from there.

Probably the best way to approach it is put the mboot bootloader in the 128k and the main application (MicroPython) all in external XIP flash.  Then mboot needs to be modified so that it memory maps the XIP before jumping to the application.

Alternatively, build a very small version of MicroPython that fits in the 128k flash.

**Reply by LAPSyLAB** [reply]
Hello, thanks for the answer... I'd love to try first to build small version that actually works.
Any size per module overview to see which modules should be left out ?
 
But I'm still curious how can I get any feedback from Micropython what is going on ?

Thanks.

**Comment by jahr** [comment]
Hello, I've tried to create the board configuration for STM32H750B_DK using mboot as @dpgeorge suggested. Though I was successful with other H750 based boards ([WEACT_750](https://github.com/jahr/WEACT_H750) and [FK750M3](https://github.com/jahr/FK750M3)), configuration for STM32H750B_DK doesn't work. You may check it [here](https://github.com/jahr/STM32H750B_DK). Mboot works but downloading firmware is very slow (comparing to the other boards), so there is probably a problem with QSPI memory configuration. The firmware itself doesn't communicate so it probably doesn't work at all. You may try to play with it, perhaps you find something I overlooked ;-)

**Reply by LAPSyLAB** [reply]
Hello, 

thanks for your help... I've based my configuration on your projects and eventually we will solve those problems...

But it would be much easier if we would have some kind of debug possibility, to check what is going on. 
I was curious if it is possible to add some debug prints on USART3, since we already have it as a VCOM port on PC.

My findings and plans :
- first, maybe I'd try to make the project smaller (with fewer modules) to see if it works in internal Flash.
- then I'd try with QSPI flash. It seems that STM32CubeProgrammer has some plugins that allow flashing of QSPI flashes - so maybe this could be done directly. I've tried it, and it seems that it puts something into QSPI flash - although I'm not sure if all settings are correct. But I need to explore further how that two-part setup should work. I've flashed the first part in internal flash and the second in QSPI flash, but it doesn't seem to work. Maybe we are near, maybe we are far from the solution - not sure. That's why I'm after some debugging or tracing options.

Regards,
Rob.
